File: URIS2023-AgilIS-Tim10-main/URIS2023_TempateProject-main/src/app/models/acceptance-criteria.ts
import { BacklogItem } from "./backog-item"

export class AcceptanceCriteria {
   acceptanceCriteriaId: string
   acceptanceCriteriaText: string
   backlogItemId: string
   backlogItem: BacklogItem
}

export class AcceptanceCriteriaCreation {
   acceptanceCriteriaText: string
   backlogItemId: string

   constructor(acceptanceCriteriaText,backlogItemId ){
    this.acceptanceCriteriaText=acceptanceCriteriaText;
    this.backlogItemId =backlogItemId
   }
}

export class AcceptanceCriteriaUpdate {
    acceptanceCriteriaId: string
    acceptanceCriteriaText: string

    constructor(acceptanceCriteriaId, acceptanceCriteriaText ){
      this.acceptanceCriteriaId =acceptanceCriteriaId
      this.acceptanceCriteriaText=acceptanceCriteriaText;
       }

 }

File: URIS2023-AgilIS-Tim10-main/BacklogItem_MicroService/BacklogItem_MicroService/Models/Entities/AcceptanceCriteria.cs
using System.ComponentModel.DataAnnotations;

namespace BacklogItem_MicroService.Models.Entities
{
    public class AcceptanceCriteria
    {
        [Key]
        public Guid AcceptanceCriteriaId { get; set; }
        [Required]
        [MaxLength(200)]
        public string AcceptanceCriteriaText { get; set; }
        
        public Guid BacklogItemId { get; set; }
        public BacklogItem BacklogItem { get; set; }
    }
}
